orbit/pkg/update/flag_runner.go

```diff
@@ -126,6 +126,160 @@ func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
 	return true, nil
 }
 
+// ExtensionRunner is a specialized runner to periodically check and update flags from Fleet
+// It is designed with Execute and Interrupt functions to be compatible with oklog/run
+//
+// It uses an OrbitClient, along with ExtensionUpdateOptions and updateRunner to connect to Fleet
+type ExtensionRunner struct {
+	configFetcher OrbitConfigFetcher
+	opt           ExtensionUpdateOptions
+	cancel        chan struct{}
+	updateRunner  *Runner
+}
+
+// ExtensionUpdateOptions is options provided for the extensions fetch/update runner
+type ExtensionUpdateOptions struct {
+	// CheckInterval is the interval to check for updates
+	CheckInterval time.Duration
+	// RootDir is the root directory for orbit state
+	RootDir string
+}
+
+// NewExtensionConfigUpdateRunner creates a new runner with provided options
+// The runner must be started with Execute
+func NewExtensionConfigUpdateRunner(configFetcher OrbitConfigFetcher, opt ExtensionUpdateOptions, updateRunner *Runner) *ExtensionRunner {
+	return &ExtensionRunner{
+		configFetcher: configFetcher,
+		opt:           opt,
+		cancel:        make(chan struct{}),
```

Comment:
Should this be a buffered `chan` like in https://github.com/fleetdm/fleet/blob/main/orbit/pkg/update/runner.go#L72-L74? Maybe that's not necessary with the `close` approach used in `Interrupt`?

Reply:
yep! we call `close` in `Interrupt` just like in the `FlagRunner`